File: apps/ios/Barns/BarnsApp/Domain/UseCases/Support/GetConsultationDraftUseCase.swift
/// Returns the user's current local consultation draft, if any.
struct GetConsultationDraftUseCase {
    private let repository: ConsultationDraftRepository

    init(repository: ConsultationDraftRepository) {
        self.repository = repository
    }

    func execute() async throws -> ConsultationDraft? {
        try await repository.currentDraft()
    }
}

File: apps/android/app/src/main/java/com/barns/app/domain/usecase/support/GetConsultationDraftUseCase.kt
package com.barns.app.domain.usecase.support

import com.barns.app.domain.model.ConsultationDraft
import com.barns.app.domain.repository.ConsultationDraftRepository

/**
 * Returns the user's current local consultation draft, if any.
 */
class GetConsultationDraftUseCase(
    private val repository: ConsultationDraftRepository,
) {
    suspend fun execute(): ConsultationDraft? = repository.currentDraft()
}
